python/lsst/cp/pipe/linearity.py

```diff
@@ -354,7 +424,20 @@ def run(self, inputPtc, dummy, camera, inputDims):
             linearizer.fitParams[ampName] = np.array(polyFit)
             linearizer.fitParamsErr[ampName] = np.array(polyFitErr)
             linearizer.fitChiSq[ampName] = chiSq
-
+            linearizer.linearFit[ampName] = linearFit
+            residuals = inputOrdinate[fluxMask] - modelOrdinate
```

Comment:
This line causes a build error from `tests/test_ptc.py`: "ValueError: operands could not be broadcast together with shapes (20,) (10,)".  

Ok, it looks like the `fluxMask` gets reset by the threshold test, and that marks everything as good (the `(20,)`), and `modelOrdinate` is defined using the previous definition (it's the `(10,)`).

Reply:
OK.  I need to study the testing some more.  I'll work on it, but it won't be until the week of Feb 14.

Reply:
I see what's happening here, but we need to discuss the best fix.  I had only tested it with linearity.Type='Spline', where this line works.  The modelOrdinate is different lengths for 'Spline' than for the others.  In the other fit types ('Polynomial', 'Squared', 'LookupTable'), the modelOrdinate is only as long as linearAbscissa, which is just the low flux part of the curve.  In 'Spline', linearOrdinate is increased to the full length of the data with this line:https://github.com/lsst/cp_pipe/blob/a4390a1b4bae57773cf476d011fe69b82882437d/python/lsst/cp/pipe/linearity.py#L259.
I think the spline fit is correct, but the other fits are not.  The modelOrdinate should include the full length of the data like it does in 'Spline'.  But maybe I am confused.

Reply:
Thinking about this some more, I think I see a simple fix, which I have pushed.